src/views/dialogs/DepositDialog2/AmountInput.tsx

```diff
@@ -0,0 +1,67 @@
+import { EventHandler } from 'react';
+
+import { SyntheticInputEvent } from 'react-number-format/types/types';
+import styled from 'styled-components';
+
+import { STRING_KEYS } from '@/constants/localization';
+
+import { useStringGetter } from '@/hooks/useStringGetter';
+
+import { AssetIcon } from '@/components/AssetIcon';
+import { Icon, IconName } from '@/components/Icon';
+
+export type AmountInputProps = {
+  value: string;
+  onChange: (newValue: string) => void;
+  token: { chain: string; address: string; symbol: string };
+  onTokenClick: () => void;
+};
+
+const numericValueRegex = /^\d*(?:\\[.])?\d*$/;
```

Comment:
is regex really the best way to do this? Surely we have a helper somewhere for this

Reply:
i think its the most efficient way :P